packages/frontend-v2/components/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,292 @@
+import { Box, Text } from "@chakra-ui/react";
+import { SearchAPI } from "@graduate/api-client";
+import {
+  IAndCourse2,
+  ICourseRange2,
+  IOrCourse2,
+  IRequiredCourse,
+  IXofManyCourse,
+  Major2,
+  Requirement2,
+  ScheduleCourse2,
+  Section,
+} from "@graduate/common";
+import { memo, useEffect, useState } from "react";
+import { DraggableScheduleCourse } from "../ScheduleCourse";
+
+interface SidebarProps {
+  major: Major2;
+}
+
+interface SidebarSectionProps {
+  section: Section;
+  courseData: { [id: string]: ScheduleCourse2<null> };
+  dndIdPrefix: string;
+}
+
+interface SidebarRequirementProps {
+  requirement: Requirement2;
+  courseData: { [id: string]: ScheduleCourse2<null> };
+  dndIdPrefix: string;
+}
+
+// This was moved out of the Sidebar component as it doesn't change
+// from run to run, but the dependency array in the course useEffect
+// would have to include since if it stayed in the component according
+// to the linter.
+const getRequiredCourses = (
+  requirements: Requirement2[],
+  requiredCourses: IRequiredCourse[]
+) => {
+  for (const requirement of requirements) {
+    if (requirement.type === "RANGE") {
+      continue;
+    } else if (requirement.type === "COURSE") {
+      requiredCourses.push(requirement);
+    } else if (requirement.type === "SECTION") {
+      getRequiredCourses(requirement.requirements, requiredCourses);
+    } else {
+      getRequiredCourses(requirement.courses, requiredCourses);
+    }
+  }
+};
+
+const Sidebar: React.FC<SidebarProps> = memo(({ major }) => {
+  const [courseData, setCourseData] = useState({});
+
+  useEffect(() => {
+    // console.info("QUERYING SEARCH ONCE", Date.now())
+    const requirements = major.requirementSections.reduce(
+      (courses: IRequiredCourse[], section: Section) => {
+        const requiredCourses: IRequiredCourse[] = [];
+        getRequiredCourses(section.requirements, requiredCourses);
+        return courses.concat(requiredCourses);
+      },
+      []
+    );
+    // console.info("Making", requirements.length, "requests")
+
+    const promises: Promise<ScheduleCourse2<null> | null>[] = [];
+
+    for (const requirement of requirements) {
+      promises.push(
+        // new Promise((res, rej) => {
```

Comment:
this commented stuff mimics the fetch course api client calls since ours are currently scuffed :(. you can use this segment to test with "fully loaded" courses in the sidebar